File: convert_to_usd.php
<?php
$file_path = "products.js";
$content = file_get_contents($file_path);
$prefix = "const products = ";

$start_pos = strpos($content, $prefix) + strlen($prefix);
$json_str = substr($content, $start_pos);

// remove trailing semicolon etc
$json_str = rtrim($json_str, " \n\r\t;");

$products = json_decode($json_str, true);

if (!$products) {
    die("Error parsing JSON. json_last_error_msg: " . json_last_error_msg() . "\n");
}

$updated_count = 0;
foreach ($products as &$p) {
    if (isset($p['price'])) {
        $p['price'] = round($p['price'] / 3800);
        $updated_count++;
    }
}

$new_content = $prefix . json_encode($products, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES) . ";\n";
file_put_contents($file_path, $new_content);

echo "Successfully converted prices to USD for $updated_count products.\n";

$dash = file_get_contents('dashboard.php');

$dash = preg_replace_callback("/(<h3>(.+?)<\/h3>\s*<div class=\"equipment-list\">\s*<ul>\s*)(.*?)(\s*<\/ul>\s*<\/div>)/s", function($m) use ($products) {
    $cat_name = trim($m[2]);
    if ($cat_name === 'Soccer (Football)') $cat_name = 'Football';
    $list_html = $m[3];

    $list_html = preg_replace_callback("/<li>(.*?)<span class=\"price\">.*?<\/span><\/li>/s", function($m2) use ($products, $cat_name) {
        $item_name = trim($m2[1]);
        $best_price = null;
        
        // Custom adjustments since sometimes category mismatch or keyword is a bit different
        $search_term = strtolower($item_name);
        
        foreach ($products as $p) {
            $p_cat = strtolower(trim($p['category']));
            if ($p_cat === strtolower($cat_name) || ($cat_name == 'Football' && $p_cat == 'soccer (football)')) {
                if (stripos($p['name'], $search_term) !== false) {
                    $best_price = $p['price'];
                    break;
                }
            }
        }
        
        if ($best_price !== null) {
            return "<li>" . $m2[1] . "<span class=\"price\">$" . $best_price . "</span></li>";
        }
        return $m2[0]; // fallback
    }, $list_html);

    return $m[1] . $list_html . $m[4];
}, $dash);

file_put_contents('dashboard.php', $dash);
echo "Dashboard updated.\n";
?>

File: convert_to_usd.py
import json
import re
import math

with open('products.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract json
prefix = "const products = "
start_idx = content.find(prefix) + len(prefix)
end_idx = content.rfind(";")
json_str = content[start_idx:end_idx].strip()

products = json.loads(json_str)

for p in products:
    # convert to usd
    usd = round(p['price'] / 3800)
    p['price'] = usd

new_content = prefix + json.dumps(products, indent=4) + ";\n"
with open('products.js', 'w', encoding='utf-8') as f:
    f.write(new_content)

# Now Let's map Dashboard items to products
# Let's read dashboard.php
with open('dashboard.php', 'r', encoding='utf-8') as f:
    dash = f.read()

# Simple print out just to inspect
from collections import defaultdict
cat_prods = defaultdict(list)
for p in products:
    cat_prods[p['category']].append(p)

for cat in cat_prods:
    print(f"Category: {cat}")
    for p in cat_prods[cat]:
        print(f"  {p['name']} - ${p['price']}")

